src/ORM/Association.php

```diff
@@ -764,9 +760,11 @@ public function attachTo(Query $query, array $options = []): void
         $dummy->where($options['conditions']);
         $this->_dispatchBeforeFind($dummy);
 
-        $joinOptions = ['table' => 1, 'conditions' => 1, 'type' => 1];
-        $options['conditions'] = $dummy->clause('where');
-        $query->join([$this->_name => array_intersect_key($options, $joinOptions)]);
+        $query->join([$this->_name => [
+            'table' => $options['table'],
+            'conditions' => $dummy->clause('where'),
+            'type' => $options['joinType'],
```

Comment:
Why not use join everywhere?

Reply:
How do you mean?

Reply:
My mistake, I got mixed up. This can't be `joinType` because it is a join on the query, and in the above hunk you're normalizing on `joinType` as that is the key used in `Association`.